test/Repository.js

```diff
@@ -1,9 +1,265 @@
+      }]);
+    });
+
+    it("should return custom packageConfigs", () => {
+      const repo = new Repository();
+      const customPackages = repo.asiniJson.packages = [{
+        glob: "package.json"
+      }, {
+        glob: "my-packages/*/package.json"
+      }];
+      assert.deepEqual(repo.packageConfigs, customPackages);
+    });
+  });
+
+  describe("get .packages", () => {
+    it("should return the packages", () => {
+      const repo = new Repository();
+      assert.deepStrictEqual(repo.packages, []);
+      assert.deepStrictEqual(repo.packages, repo.packages, "getter should use cached value");
+    });
+  });
+
+  describe("get .filteredPackages", () => {
+    it("should return the filteredPackages", () => {
+      const repo = new Repository();
+      assert.deepStrictEqual(repo.filteredPackages, []);
+      assert.deepStrictEqual(repo.filteredPackages, repo.filteredPackages, "getter should use cached value");
+    });
+  });
+
+  describe("get .packageGraph", () => {
+    it("should return the packageGraph", () => {
+      const repo = new Repository();
+      assert.ok(repo.packageGraph);
+      assert.ok(repo.packageGraph.constructor, "packageGraph did not have a constructor");
+      assert.equal(repo.packageGraph.constructor.name, "PackageGraph");
+      assert.deepStrictEqual(repo.packageGraph, repo.packageGraph, "getter should use cached value");
+    });
+  });
+
+  describe("buildPackageGraph()", () => {
+    it("calls expected utilities", () => {
+      const repo = new Repository();
+
+      const packages = [];
+      const packageGraph = {};
+      const filteredPackages = [];
+
+      assertStubbedCalls([
+        [PackageUtilities, "getPackages", {}, [
+          { args: [repo], returns: packages }
+        ]],
+        [PackageUtilities, "getPackageGraph", {}, [
+          { args: [packages], returns: packageGraph }
+        ]],
+        [PackageUtilities, "getFilteredPackages", {}, [
+          { args: [packages, { scope: undefined, ignore: undefined }], returns: filteredPackages }
+        ]]
+      ]);
+
+      repo.buildPackageGraph();
+    });
+  });
+
+  describe("hasCompatibleLocalAsini()", () => {
+    it("should return true when asini major version matches", () => {
+      const repo = new Repository();
+      assert.strictEqual(repo.hasCompatibleLocalAsini("500.250.0"), true);
+    });
+
+    it("should return true when asiniVersion is identical", () => {
+      const repo = new Repository();
+      assert.strictEqual(repo.hasCompatibleLocalAsini("500.0.0"), true);
+    });
+
+    it("should return false when asini major version does not match", () => {
+      const repo = new Repository();
+      assert.strictEqual(repo.hasCompatibleLocalAsini("1000.0.0"), false);
+    });
+  });
+
+  describe("hasDependencyInstalled()", () => {
+    it("should match installed dependency", () => {
+      const repo = new Repository();
+      assert.strictEqual(repo.hasDependencyInstalled("external", "^1"), true);
```

Comment:
Ah, I forgot to add the `node_modules` fixtures directory, was wondering why it worked for me locally...